lib/smooth/application/io/spi/BME280Core.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+Smooth - A C++ framework for embedded programming on top of Espressif's ESP-IDF
+Copyright 2019 Per Malmberg (https://gitbub.com/PerMalmberg)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include <smooth/core/util/ByteSet.h>
+#include <smooth/application/io/spi/BME280Core.h>
+
+using namespace smooth::core;
+
+namespace smooth::application::sensor
+{
+    static const char TAG[] = "BME280Core:";
+
+    // Constructor
+    BME280Core::BME280Core()
+    {
+    }
+
+    // Get config_sensor datagram that contains register addresses and data that will be used to configure the BME280
+    void BME280Core::get_configure_sensor_datagram(std::vector<uint8_t>& datagram,
```

Comment:
This might be inherited from my old code, but lets improve it. Using a vector to pass six uint8_t seems overkill. Lets make `datagram` into a `std::array<uint8_t, 6>`

Reply:
The I2CMasterDevices expects a vector for it's writes.
I could modify io/i2c/BME280.cpp use the std::array datagram and copy to a vector that then can be used as the write.
```
io/i2c/BME280.cpp (existing - not modified yet)

bool BME280::configure_sensor(BME280Core::SensorMode mode,
                                  BME280Core::OverSampling humidity,
                                  BME280Core::OverSampling pressure,
                                  BME280Core::OverSampling temperature,
                                  BME280Core::StandbyTimeMS standby,
                                  BME280Core::FilterCoeff filter_coeff,
                                  BME280Core::SpiInterface spi_interface)
    {
        std::vector<uint8_t> datagram;

        bme280_core.get_configure_sensor_datagram(datagram,
                                                  mode,
                                                  humidity,
                                                  pressure,
                                                  temperature,
                                                  standby,
                                                  filter_coeff,
                                                  spi_interface);

        return write(address, datagram, true);
    }
```
Is that what you prefer or leave implenation in BME280Core as vector?

Reply:
Oh, right. Silly me.

No, let it be as-is.